### PR TITLE
Port HTTPRequest gzip compression to 3.4

### PR DESCRIPTION
**Pull request by tavurth**
Tests are finally passing 🎉 

https://github.com/godotengine/godot/pull/48581

_Edit:_ Backport of #38944

### CODE REVIEW

**Comment by Wavesonics** [comment]
Nice work, it looks good to me.

**Comment by Wavesonics** [comment]
Nice, it looks good to me!

File: scene/main/http_request.h [hunk 1]
```diff
@@ -51,6 +51,7 @@ class HTTPRequest : public Node {
 		RESULT_SSL_HANDSHAKE_ERROR,
 		RESULT_NO_RESPONSE,
 		RESULT_BODY_SIZE_LIMIT_EXCEEDED,
+		RESULT_BODY_DECOMPRESS_FAILED,
```
Comment (akien-mga) [inline scene/main/http_request.h:54]: You should also bind it in `_bind_methods`.

Comment (tavurth) [reply scene/main/http_request.h:54]: @akien-mga thank you! Updated the PR.

**Comment by tavurth** [comment]
Ping :)

**Comment by tavurth** [comment]
@Calinou sorry to ping you on this, is there something else I should do to push for this in the 3.4 release?

Exited to see this feature speeding up my API calls :)

**Comment by Calinou** [comment]
> @Calinou sorry to ping you on this, is there something else I should do to push for this in the 3.4 release?
> 
> Excited to see this feature speeding up my API calls :)

Akien is currently on vacation, so large pull requests won't be merged for now.

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Comment by Wavesonics** [comment]
I can put some time in on the master implementation to get it fixed up

**Comment by RPicster** [comment]
Will this feature be looked into again?
It would be great to have in 3.x.
Could it be eventually solved by not making it automatic but by a parameter?

**Comment by tavurth** [comment]
@RPicster if we have a working master implementation then this should be pretty trivial to port back to 3.x again.

**Comment by Wavesonics** [comment]
I'm taking a look at what needs to be done here,

- `Some platform/HTTPClient might not allow those headers (HTML5)` can you elaborate? Is this a problem with the web export? What would a solution look like? Disable the option to include the header on that export or something?
- `Response is not decompressed when saved to file.` easy enough, I can implement that
- `We need facilities to decompress a stream (i.e. not decompress all in memory but in chunks to file).` I *think* this should be do-able using the new stream decompression I added for this already, I'll look into it.

Other than these 3 issues is there anything else that is needed to consider this fixed?

**Comment by Wavesonics** [comment]
#55946 addresses the 2nd two points listed above, we can now stream decompress a file to another file (_though that is not exposed through GDScript yet._) and that is then used to decompress the downloaded to file option in `HttpRequest`.

I don't have any idea how we deal with HTML5 as a platform in Godot, so maybe someone could clue me in. It sort of sounds like on HTML5 exports, we should just turn off the "Accept Gzip header" option under the hood or something?

**Pull request merged**